guvnor-m2repo-editor/guvnor-m2repo-editor-api/src/main/java/org/guvnor/m2repo/preferences/ArtifactRepositoryPreference.java

```diff
@@ -24,20 +24,68 @@
 @WorkbenchPreference(identifier = "ArtifactRepositoryPreference", bundleKey = "ArtifactRepositoryPreference.Label")
 public class ArtifactRepositoryPreference implements BasePreference<ArtifactRepositoryPreference> {
 
-    @Property(bundleKey = "ArtifactRepositoryPreference.DefaultM2RepoDir")
-    private String defaultM2RepoDir;
+    @Property(bundleKey = "ArtifactRepositoryPreference.GlobalM2RepoDir")
+    private String globalM2RepoDir;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.GlobalM2RepoDirEnabled")
+    private boolean globalM2RepoDirEnabled;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.WorkspaceM2RepoDir")
+    private String workspaceM2RepoDir;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.WorkspaceM2RepoDirEnabled")
+    private boolean workspaceM2RepoDirEnabled;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.DistributionManagementM2RepoDirEnabled")
+    private boolean distributionManagementM2RepoDirEnabled;
 
     @Override
     public ArtifactRepositoryPreference defaultValue(final ArtifactRepositoryPreference defaultValue) {
-        defaultValue.defaultM2RepoDir = "repositories/kie";
+        defaultValue.globalM2RepoDir = "repositories/kie/global";
+        defaultValue.workspaceM2RepoDir = "repositories/kie/workspaces";
+        defaultValue.globalM2RepoDirEnabled = true;
+        defaultValue.workspaceM2RepoDirEnabled = false;
+        defaultValue.distributionManagementM2RepoDirEnabled = true;
```

Comment:
@manstis the problem here is that I chose a not so happy name =S. The only thing I wanted to do here is to replicate some behaviour that had this code before to "not to use distribution management section". The name is what is wrong, it should be distributionManagementEnabled, and it should enable or disable the posibility that GuvnorM2Repository to deploy an artifact in the distributionManagement setted into the pom.